datasets/za/pmg_legislators/crawler.py

```diff
@@ -0,0 +1,244 @@
+            person.add("address", value)
+
+    for membership in person_data.pop("memberships", []):
+        crawl_membership(context, person, membership, organizations, events)
+
+
+def crawl_membership(
+    context: Context,
+    entity: Entity,
+    membership: dict,
+    organizations: Dict[str, str],
+    events,
+) -> Optional[str]:
+    org_id = membership.get("organization_id")
+    org_name = organizations.get(org_id)
+    if not org_name:
+        # context.log.error(f"Organization with ID {org_id} not found.")
+        return None
+
+    role = membership.get("role")
+    if role is None:
+        # context.log.error("Role is missing in membership data.")
+        return None
+
+    # Creating the position property
+    position_property = post_summary(
+        org_name, role, [membership.get("start_date")], [membership.get("end_date")], []
+    )
+    entity.add("position", position_property)
+
+    # context.log.info(f"Processed membership for person ID: {person_id}")
+    position_label = f"{role.title()} of the {org_name}"
+
+    # Check if the role matches the positions of interest
+    if not any(re.findall(regex, position_label) for regex in POSITIONS_OF_INTEREST):
+        # context.log.info(f"Skipping role {role} not of interest.")
+        return None
+
+    # Initialize the topics list
+    topics = []
+
+    # Check if the role matches GOV_POSITIONS or LEG_POSITIONS
+    if any(re.findall(regex, position_label) for regex in GOV_POSITIONS):
+        topics.append("gov.national")
+
+    if any(re.findall(regex, position_label) for regex in LEG_POSITIONS):
+        topics.append("gov.legislative")
+
+    # If neither matches, we can skip this role
+    if not topics:
+        return None
+
+    position = h.make_position(
+        context,
+        position_label,
+        country="za",
+        topics=topics,
+    )
+
+    # Always PEP because filtered by known position label patterns
+    categorisation = categorise(context, position, is_pep=True)
+
+    period_id = membership.get("legislative_period_id")
+    period = events.get(period_id, {})
+    role = membership.pop("role", None)
+    role = role or period.get("name")
+
+    # If the role is not "member", we might want to log a warning.
+    # if role != "member":
+    #     context.log.warning("Unexpected role", role=role)
+
+    starts = [membership.get("start_date"), period.get("start_date")]
+    ends = [membership.get("end_date"), period.get("end_date")]
+    position_property = post_summary(org_name, role, starts, ends, [])
+
+    # Creating and emitting the Occupancy entity
+    occupancy = h.make_occupancy(
+        context,
+        entity,
+        position,
+        no_end_implies_current=False,
+        start_date=membership.get("start_date") or period.get("start_date"),
+        end_date=membership.get("end_date") or period.get("end_date"),
+        birth_date=membership.get("birth_date"),
+        death_date=membership.get("death_date"),
```

Comment:
a membership has a birth/death date?

it looks like this isn't working correctly - Collins Chabane died in 2015 so they're outside the 5 year window

```
"id": "Q1109753",
  "caption": "Ohm Collins Chabane",
  "schema": "Person",
  "properties": {
    "position": [
      "National Assembly (Member, 2014-2015)",
      "National Assembly (Member, 2009-2014)",
      "National Executive (Minister in The Presidency: Performance Monitoring and Evaluation as well as Administration in the in the Presidency, -2014)",
      "African National Congress (ANC) (Member)",
      "ANC Constituency Office (149): Vereeniging Regional Office (Constituency Contact, 2014-2014)",
      "National Assembly (Member-elect, 2014-2014)",
      "ANC Constituency Office (149): Vereeniging Regional Office (Constituency Contact, -2014)",
      "National Executive (Minister of Public Service and Administration, 2014-2015)",
      "African National Congress National Election List 2014 (10th Candidate, 2014-2014)"
    ],
    "name": [
      "Ohm Collins Chabane"
    ],
    "country": [
      "za"
    ],
    "title": [
      "Mr"
    ],
    "firstName": [
      "Ohm Collins"
    ],
    "wikidataId": [
      "Q1109753"
    ],
    "lastName": [
      "Chabane"
    ],
    "deathDate": [
      "2015-03-15"
    ],
```

Maybe we want to use person.get(birth/death date?

Also maybe .get() doesn't crash if it's missing? Perhaps use .pop() to discover bugs like this more easily?